Sources/Sentry/SentryANRTracker.m

```diff
@@ -89,7 +89,13 @@ - (void)detectANRs
 
             if (reported) {
                 SENTRY_LOG_WARN(@"ANR stopped.");
-                [self ANRStopped];
+
+                // The ANR stopped, don't block the main thread with calling ANRStopped listeners.
+                // While the ANR code reports an ANR and collects the stack trace, the ANR might
+                // stop simultaneously. In that case, the ANRs stack trace would contain the
+                // following code running on the main thread. To avoid this, we offload work to a
+                // background thread.
+                [self.dispatchQueueWrapper dispatchAsyncWithBlock:^{ [self ANRStopped]; }];
```

Comment:
This will also delay the suspension of threads to gather stacktraces. It might be better to separate the listeners and to collect the stack traces immediately, and only report the app state update from an async context.

Reply:
We don't getter stacktraces when reporting the the app hanging stopped

Reply:
Exactly what @brustolin said, but thanks for raising the concern, @armcknight.